adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -568,10 +573,11 @@ def disconnect(self):
         if resp[0][0] != 1:
             raise OSError("Failed to disconnect")
 
-    def connect(self, secrets):
-        """Connect to an access point using a secrets dictionary
-        that contains a 'ssid' and 'password' entry"""
-        self.connect_AP(secrets["ssid"], secrets["password"])
+    def connect(self, ssid, password=None, timeout=10):
+        """Connect to an access point with given name and password."""
+        if isinstance(ssid, dict):  # secrets
+            ssid, password = ssid["ssid"], ssid["password"]
```

Comment:
@anecdata didn't think about `open`. This should be
```suggestion
            ssid, password = ssid["ssid"], ssid.get("password")
```
to allow `password` to not exist

Reply:
good catch